packages/data/src/world/mq/ganon_castle_mq.yml

```diff
@@ -35,92 +39,254 @@
   events:
-    GANON_TRIAL_FOREST: "soul_enemy(SOUL_ENEMY_STALFOS) && (can_play_time || can_play_elegy) && can_use_light_arrows"
+    GANON_CASTLE_MQ_FOREST_WIND_FROZEN_EYE: "has_fire"
+    GANON_CASTLE_MQ_FOREST_WIND_OPEN_EYE: "can_hit_triggers_distance"
+    GANON_MQ_FOREST_SWITCH: "can_play_time || can_play_elegy"
   locations:
-    "MQ Ganon Castle Forest Trial Key": "can_hookshot"
-    "MQ Ganon Castle Forest Trial First Chest": "soul_enemy(SOUL_ENEMY_STALFOS) && has_weapon && can_hit_triggers_distance"
-    "MQ Ganon Castle Forest Trial Second Chest": "soul_enemy(SOUL_ENEMY_STALFOS) && has_weapon && has_fire"
-    "MQ Ganon Pot Forest End 1": "soul_enemy(SOUL_ENEMY_STALFOS) && (can_play_time || can_play_elegy)"
-    "MQ Ganon Pot Forest End 2": "soul_enemy(SOUL_ENEMY_STALFOS) && (can_play_time || can_play_elegy)"
-"Ganon Castle Fire":
+    "MQ Ganon Castle Forest Trial Second Chest": "event(GANON_CASTLE_MQ_FOREST_WIND_FROZEN_EYE)"
+"Ganon Castle Forest Trial Ending Room":
   dungeon: Ganon
+  exits:
+    "Ganon Castle Forest Trial Wind Room Ending Side": "true"
   events:
-    GANON_TRIAL_FIRE: "can_use_light_arrows && has_tunic_goron_strict && (can_longshot || has_hover_boots) && silver_rupees_ganon_fire"
+    GANON_TRIAL_FOREST: "can_use_light_arrows"
+  locations:
+    "MQ Ganon Pot Forest End 1": "true"
+    "MQ Ganon Pot Forest End 2": "true"
+"Ganon Castle Fire Trial Entrance Ledge":
+  dungeon: Ganon
+  exits:
+    "Ganon Castle Main Room": "true"
+    "Ganon Castle Fire Trial Ending Ledge": "has_tunic_goron_strict && (can_longshot || has_hover_boots)"
   locations:
     "MQ Ganon Castle SR Fire Under Pillar": "has_tunic_goron && can_lift_gold"
     "MQ Ganon Castle SR Fire Center-Left": "has_tunic_goron_strict"
     "MQ Ganon Castle SR Fire Back-Left": "has_tunic_goron_strict"
     "MQ Ganon Castle SR Fire High Above Lava": "has_tunic_goron"
     "MQ Ganon Castle SR Fire Front-Left": "has_tunic_goron_strict"
-    "MQ Ganon Pot Fire End 1": "has_tunic_goron_strict && (can_longshot || has_hover_boots) && silver_rupees_ganon_fire"
-    "MQ Ganon Pot Fire End 2": "has_tunic_goron_strict && (can_longshot || has_hover_boots) && silver_rupees_ganon_fire"
-"Ganon Castle Water":
+"Ganon Castle Fire Trial Ending Ledge":
   dungeon: Ganon
+  exits:
+    "Ganon Castle Fire Trial Entrance Ledge": "has_tunic_goron_strict && longshot_anywhere"
+    "Ganon Castle Fire Trial Ending Room": "silver_rupees_ganon_fire"
+  locations: #Needs checking.
+    "MQ Ganon Castle SR Fire Under Pillar": "longshot_anywhere && has_tunic_goron_strict && can_lift_gold"
+    "MQ Ganon Castle SR Fire Center-Left": "longshot_anywhere && has_tunic_goron_strict"
+    "MQ Ganon Castle SR Fire Back-Left": "longshot_anywhere && has_tunic_goron_strict"
+    "MQ Ganon Castle SR Fire High Above Lava": "longshot_anywhere && has_tunic_goron_strict"
+    "MQ Ganon Castle SR Fire Front-Left": "longshot_anywhere && has_tunic_goron_strict"
```

Comment:
From preliminary testing, none of these require strict, but there will be no way to tell for sure until we can directly come out of that door so that the cycles are in their initial state.

Reply:
Notes added.